traffic_ops/traffic_ops_golang/login/login.go

```diff
@@ -21,26 +21,83 @@ package login
 
 import (
 	"bytes"
+	"database/sql"
 	"encoding/json"
 	"errors"
 	"fmt"
-	"github.com/dgrijalva/jwt-go"
-	"github.com/lestrrat-go/jwx/jwk"
+	"html/template"
 	"net/http"
 	"net/url"
 	"path/filepath"
 	"time"
 
 	"github.com/apache/trafficcontrol/lib/go-log"
 	"github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/lib/go-rfc"
 	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
 	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/auth"
 	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/config"
 	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/tocookie"
 
+	"github.com/dgrijalva/jwt-go"
+	"github.com/google/uuid"
 	"github.com/jmoiron/sqlx"
+	"github.com/lestrrat-go/jwx/jwk"
 )
 
+type emailFormatter struct {
+	From rfc.EmailAddress
+	To rfc.EmailAddress
+	InstanceName string
+	ResetURL rfc.URL
+	Token string
+}
+
+const instanceNameQuery = `
+SELECT value
+FROM parameter
+WHERE name='tm.instance_name' AND
+      config_file='global'
+`
+const userQueryByEmail = `SELECT COUNT(*)::int::bool FROM tm_user WHERE email=$1`
+const setTokenQuery = `UPDATE tm_user SET token=$1 WHERE email=$2`
+var resetPasswordEmailTemplate = template.Must(template.New("Password Reset Email").Parse("From: {{.From}}\r"+`
+To: {{.To}}`+"\r"+`
```

Comment:
I wonder if we could make a common email layout template that we can leverage in all our emails.

Since there are multiple places that will need to send emails we could have a email layout template that has the From/TO/Subject, HTML set up and then the users of it can just worry about setting From/TO/Subject values as well as the body and style (unless this is common across all emails as well)

Reply:
Maybe. Sounds complex, would need to contain nested templates that unravel to suit the needs of the caller. In any case I think that's outside of the scope of this PR, as it nearly constitutes a system all its own.

Reply:
Don't think it would be too complex would just need to include the base template as you  render it. And yeah might be out of scope but would be a good time to have the conversation as we add emailing templating to the system

Reply:
Fine leaving the implementation as is. Was just a suggestion :)

Reply:
Okay, I think maybe I misunderstood. You're saying there's one master template that'd look something like:

```go
const EmailTemplate = "To: {.To}\r"+`
From: {.From}`+"\r"+`
Subject: {.Subject}`+"\r"+`
{if .CC}CC: {.CC}`+"\r\n{endif}\r"+`
<!DOCTYPE html>
<html lang="enUS">
<head>
<title>{.Subject}</title>
<meta charset="utf-8"/>
<style>/* Maybe some styling here */</style>
</head>
<body>
%s <-- Embed template? -->
</body>
</html>
```

and then each other thing fills out the `<body>` for itself?

Reply:
yeah! something like that :)